pkg/hubble/relay/pool/manager.go

```diff
@@ -319,8 +319,7 @@ func (m *PeerManager) connect(p *peer, ignoreBackoff bool) {
 	m.opts.log.WithFields(logrus.Fields{
 		"address": p.Address,
 	}).Debugf("Connecting peer %s...", p.Name)
-	//FIXME: provide hostname to ClientConn
-	conn, err := m.opts.clientConnBuilder.ClientConn(p.Address.String(), "")
+	conn, err := m.opts.clientConnBuilder.ClientConn(p.Address.String(), p.TLSServerName)
```

Comment:
Should we throw a warning if `p.TLSEnabled=true` but hubble-relay was not configured with any certs for TLS for peers?

Reply:
It would be nice to throw a warning but in practice it's a little difficult as the pool manager is not aware whether relay is configured with certs for TLS or not (the `ClientConnBuilder` is set when instantiating a new `Server` and at this point the `Server` is not aware if peers are TLS enabled or not and the pool manager is not aware of how `ClientConn()` establishes the connection (it's an interface)).
TLDR: it would require quite some changes to implement. If you feel it's a must have, I suggest to open an issue and address this later.

Reply:
We can check `m.opts.clientConnBuilder.TLSConfig` right? Alternatively, `ClientConn()` could return an error when it has a TLS config while called with an empty server name.

Reply:
> We can check m.opts.clientConnBuilder.TLSConfig right?

I'm not really sure what you are trying to imply here.  The current code uses the `GRPCClientConnBuilder` as the implementation of the `ClientConnBuilder` interface. This implementation opts to connect using TLS if a TLS config is provided and uses plaintext otherwise.

I think that what @gandro is pointing out is that the peer service already reports whether a peer is configured with TLS or not (see `Peer.TLSEnabled`) but this information is discarded by the pool manager which, by abstraction, knows nothing about how the connection to peers is established. In order to change this, the `ClientConnBuilder` interface should be modified such that the caller defines whether TLS is required or not.

> Alternatively, ClientConn() could return an error when it has a TLS config while called with an empty server name.

I think that this is what happens in practice currently as the TLS handshake will fail as the Server Name will not match the certificate's.

Reply:
Let's follow up in #13087